skills.md

```diff
@@ -0,0 +1,138 @@
+                    <th> {{ver}} </th>
+                {%- endfor -%}
+            {%- endfor -%}
+            </tr>
+        {%- endif -%}
+
+            {%- for author in site.authors -%}
+
+                {%- comment -%} Check if skill_name is found inside the author skill list, continue otherwise {%- endcomment -%}
+                {%- assign skill_found = false -%}
+                {%- for a_skill in author[stype] -%}
+                    {%- if a_skill.name == skill_name -%}
+                        {%- assign skill_found = a_skill -%}
+                        {%- break -%}
+                    {%- endif -%}
+                {%- endfor -%}
+                {%- unless skill_found -%}
+                    {%- continue -%}
+                {%- endunless -%}
+
+                <tr>
+                    <td><a href="{{author.url}}">{{author.name}}</a></td>
+                {%- for type in self_want -%}
+                    {%- if skill.vers -%}
+                        {%- for ver in skill.vers -%}
+                            {%- assign score = 0 -%}
+
+{%- comment -%} returns score for matching (skill_name, ver). If no matching version is available but
+                there is a versionless entry, such score will be used {%- endcomment -%}
+                            {%- assign skill_list = author[stype] -%}
+                            {%- assign skill_found = false -%}
+                            {%- assign skill_found_generic = false -%}
+                            {%- for a_skill in skill_list -%}
+                                {%- if a_skill.name == skill_name -%}
+                                    {% if a_skill.ver%}
+                                        {%- for a_ver in a_skill.ver -%}
+                                            {%- if a_ver == ver -%}
+                                                {%- assign skill_found = a_skill -%}
+                                                {%- break -%}
+                                            {%- endif -%}
+                                        {%- endfor -%}
+                                    {%- else -%}
+                                        {%- assign skill_found_generic = a_skill -%}
+                                    {%- endif -%}
+                                {%- endif -%}
+                                {%- if skill_found -%}
+                                    {%- break -%}
+                                {%- endif -%}
+                            {%- endfor -%}
+                            {%- if skill_found or skill_found_generic -%}
+                                {%- unless skill_found -%}
+                                    {%- assign skill_found = skill_found_generic -%}
+                                {%- endunless -%}
+                                {%- assign score = skill_found[type] -%}
+                            {%- endif -%}
+{%-comment-%} This is disabled but we will keep it here for backup reason
```

Comment:
Is this never actually called or is there some extremely rare condition where this is needed?
Also,  I understand this does what the previous block does, but without "generalizing" a versionless skill - is that correct?

Reply:
Since it is between comment tags, it's actually never even parsed.

Is an old and less generalized version which I was sure it worked and just forgot to delete . My bad!

Reply:
Totally missed the endcomment tag and thought this was a line comment. Yay.

Reply:
By no means this comment style is confusing, right?

Trivia: HTML-style comment are blatantly ignored by liquid, and therefore treated as normal text since it's processing happens before parsing the markdown, which happens before the HTML transposition.